src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1478,7 +1478,8 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
         os.remove(temp_yaml_path)
 
     print()
-    normalized_fqdn = mc.fqdn.replace('.', '-')
+    fqdn = mc.fqdn if mc.fqdn is not None else mc.private_fqdn
```

Comment:
👍  LGTM, nice, quick question: do we have any `msdn` doc which list these `objects`

Reply:
those properties are coming from AKS rest API, e.g. https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/get

Reply:
+1 /LGTM